Nos/Service/NamesAPI.swift

```diff
@@ -51,15 +51,33 @@ class NamesAPI {
         throw Error.unexpected
     }
 
+    /// Verifies that a given username is free to claim in nos.social
     func verify(username: String, keyPair: KeyPair) async throws -> Bool {
+        try await verify(
+            username: username,
+            host: verificationURL,
+            keyPair: keyPair,
+            valueWhenNotFound: true
+        )
+    }
+
+    /// Verifies that a given username_at_host NIP-05 can be connected to the keyPair
+    ///
+    /// - Parameter valueWhenNotFound: What to return if the server returns 404
+    func verify(
+        username: String,
+        host: URL,
+        keyPair: KeyPair,
+        valueWhenNotFound: Bool = false
+    ) async throws -> Bool {
         let request = URLRequest(
-            url: verificationURL.appending(queryItems: [URLQueryItem(name: "name", value: username)])
+            url: host.appending(queryItems: [URLQueryItem(name: "name", value: username)])
         )
         let (data, response) = try await URLSession.shared.data(for: request)
         if let response = response as? HTTPURLResponse {
             let statusCode = response.statusCode
             if statusCode == 404 {
-                return true
+                return valueWhenNotFound
```

Comment:
Maybe instead of having the caller indicate that they want this bool value to be flipped it would be less confusing if they just flipped it themselves at the call site?

Reply:
But the caller won't know if it was a 404 or that the server didn't return an unexpected response.
A different approach would be to return a enum value so the called would know what was the server response.